src/network/networkd-link.c

```diff
@@ -126,6 +129,9 @@ static bool link_dhcp4_enabled(Link *link) {
         if (link->network->bond)
                 return false;
 
+        if (link->iftype == ARPHRD_CAN)
+                return false;
+
         if (STRPTR_IN_SET(link->kind, "can", "vcan", "vxcan"))
                 return false;
```

Comment:
maybe merge this into a new helper:

```c
static inline bool LINK_IS_CAN(Link *l) {
        assert(l);
        if (l->iftye == ARPHRD_CAN)
                return true;
        return STRPTR_IN_SET(link->kind, "can", "vcan", "vxcan");
}
```

Reply:
Why is the `link->kind` check still necessary? AFAIK, every CAN driver has `iftype == ARPHRD_CAN`.

Reply:
Indeed. I will submit follow-up PR. Thanks.

Reply:
#13178.